DDF-1596 Add UI changes to handle external auth

harrison-tarr opened the pull request:
Add external authentication config to Search UI metatype
Displays a "Sign in with credentials" button that calls "/logout"
when clicked instead of a log-in form when the "externalAuthentication"
property is set.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf/284)
<!-- Reviewable:end -->

harrison-tarr commented:
@AzGoalie @brendan-hofmann @stustison @pklinef 

To hero:
1) Run PRBS
2) start and install DDF
3) after logging in, go to Search UI and ensure it displays "admin" as the user in the top right
4) log out using Search UI logout, ensure it displays "sign in" and when clicked, the drop down log in form
5) go to Admin UI and relog in
6) go to standard search ui settings (in the Admin UI) and enable "external authentication"
7) refresh search ui and ensure it displays "admin"
8) log out of admin and ensure it says "sign in" and when clicked, a dropdown with a button that says "Sign In With Credentials"
9) click the button, ensuring it reloads the page and has the same "sign in" -> "sign in with credentials" dropdown

brendan-hofmann commented:
:+1: 
<!-- Reviewable comment -K1aLH4TmgCwsrp4a744:-510466975 -->

---

Reviewed 7 of 7 files at r1.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/284)
<!-- Sent from Reviewable.io -->

brendan-hofmann commented:
Heroed successfully. :shipit:

stustison commented:
Can we also update the documentation for the IdP to instruct administrators to hit this checkbox if they configure /search to use the IDP handler.

pklinef commented on catalog/ui/search-ui/standard/src/main/webapp/templates/menu/menubarLoginExternal.handlebars:
```diff
@@ -0,0 +1,16 @@
+ **/
+ --}}
+<div class="logout-container">
+    <a href="#" class="btn btn-medium btn-primary btn-loginExternal">Sign In With Credentials</a>
```
Why not just "Sign In" like most pages?

brendan-hofmann replied:
Well, technically they're already signed in as anonymous/guest, right?

stustison replied:
Yeah, they are. That is why I thought it was a little weird to say "Sign In", but I might be able to be convinced otherwise.

pklinef replied:
The end user does not know that they are technically signed in.  I was expecting something that would better match the Amazon approach.

AzGoalie commented:
Reviewed 7 of 7 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/284)
<!-- Sent from Reviewable.io -->

harrison-tarr commented:
@stustison @pklinef @brendan-hofmann have you all reached a design consensus? is the current "Sign In With Credentials" okay? do you want it changed?

brendan-hofmann commented:
[[selfie-placeholder-0]]

brendan-hofmann commented:
![selfie-0](http://i.imgur.com/Vytt1WZ.png)

pklinef commented:
Beyond being conciseness, I also like the symmetry of "Sign In" for guest relative to "Sign Out" once they do log in as a user.

stustison commented:
Talked to @pklinef and we decided to go with: when using external auth, do not show "Sign In" in the upper right, but always display the name in the token (anonymous in this particular case) and then you can just use the log out button as it already exists.

brendan-hofmann commented:
Re-heroed, still passes, shows `Anonymous` and `Log Out` now when external auth is checked in the search ui config.

stustison commented on catalog/ui/search-ui/standard/src/main/webapp/templates/menu/menubarLoginExternal.handlebars:
```diff
@@ -0,0 +1,16 @@
+{{!--
```
Should this file just be removed now that we are showing "anonymous" and a "log out" button instead?

stustison commented on catalog/ui/search-ui/standard/src/main/webapp/css/searchPage.css:
```diff
@@ -988,6 +988,10 @@ button.ui-multiselect {
     display: block;
 }
 
+.btn-loginExternal {
```
This is no longer required now, right?

stustison commented on catalog/ui/search-ui/standard/src/test/js/wd/externalLoginMenu.js:
```diff
@@ -0,0 +1,16 @@
+
+describe('external authentication Login', function() {
+    shared.setup(this);
+    it("should have 'Sign In With Credentials' button", function() {
```
This description needs changing?

harrison-tarr commented:
Review status: 3 of 6 files reviewed at latest revision, 4 unresolved discussions.

---

<sup>**[catalog/ui/search-ui/standard/src/test/js/wd/externalLoginMenu.js, line 9 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/284#-K1_rbA7hoA53NzQmpg--r3-9)** ([raw file](https://github.com/codice/ddf/blob/510a9bb2a12809a774e42f2f88c34d68b05b42d2/catalog/ui/search-ui/standard/src/test/js/wd/externalLoginMenu.js#L9)):</sup>
Fixed.
<!-- Reviewable comment -K2DLCLHw99uNvkBEzjh:1758359711 -->

---

<sup>**[catalog/ui/search-ui/standard/src/main/webapp/css/searchPage.css, line 991 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/284#-K1_rbA7hoA53NzQmpfw-r3-991)** ([raw file](https://github.com/codice/ddf/blob/510a9bb2a12809a774e42f2f88c34d68b05b42d2/catalog/ui/search-ui/standard/src/main/webapp/css/searchPage.css#L991)):</sup>
Removed.
<!-- Reviewable comment -K2DLDvmU3WgOKSXCKLV:1740281843 -->

---

<sup>**[catalog/ui/search-ui/standard/src/main/webapp/templates/menu/menubarLoginExternal.handlebars, line 1 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/284#-K1_rbA7hoA53NzQmpfz-r3-1)** ([raw file](https://github.com/codice/ddf/blob/510a9bb2a12809a774e42f2f88c34d68b05b42d2/catalog/ui/search-ui/standard/src/main/webapp/templates/menu/menubarLoginExternal.handlebars#L1)):</sup>
Removed.
<!-- Reviewable comment -K2DLF22UnXrcguV7NAn:1740281843 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/284)
<!-- Sent from Reviewable.io -->
